evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -526,10 +619,10 @@ mod tests {
         let runner = proptest::test_runner::TestRunner::new(cfg);
 
         // ensure the storage slot is set at 10 anyway
-        let (storage_contract, _, _) = evm
+        let (storage_contract, _, _, _) = evm
```

Comment:
it seems like `ethers-rs` abigen could benefit from method call objects that encapsulate the payload, similarly to how events are handled.
It would be doable to generate supporting types for a Request/Response pattern

```rust

trait Method {
  type Response : Tokenziable;

  fn abi() -> Cow;
 }

enum Request {
    Store,
    Slot0,
    "MethodName(params)"
}

enum Response {
   Store(address),
   Slot(unit256),
   ...
}

impl Contract {
    fn execute<T:Method>(req: Method) -> Method::Response;

   fn method_name(req: impl Into<MethodName>) -> MethodName::Response;

```